### DISCUSSION TITLE
[rehype-cli] apparent inconsistency between remark-cli and programmatic use of unified

### DISCUSSION THREAD

**Discussion by Trott** [remarkjs/remark#864] [Q&A] [answered]
Based on https://github.com/remarkjs/remark/discussions/863#discussioncomment-1383476, I am trying to disable `no-auto-link-without-protocol` in `remark-preset-lint-node`.

I did it by adding this line to the `plugins` array:

```js
  ["remark-lint-no-auto-link-without-protocol", false],
```

You can see the commit here: https://github.com/nodejs/remark-preset-lint-node/pull/272/commits/360436e920eb4bc4e32038ef1ff4de14620e697a The change above is on line 75.

This works as expected with `remark -u remark-preset-lint-node input.md` but does not work programmatically with something like this:

```js
const linter = unified()
  .use(remarkParse)
  .use(presetLintNode)
  .use(remarkStringify);

const file = await read(new URL('./fixtures/ok.md', import.meta.url));
const result = await linter.process(file);
assert.strictEqual(result.messages.length, 0, reporter(result));
```

Here's `./fixtures/ok.md`:

```markdown
<fhqwhgads@example.com>
```

Using `remark-cli` and the modified preset, that file no longer gets flagged, but with the programmatic linting check shown just prior, it is still getting flagged.

Is that expected behavior? Am I doing something wrong?

**Comment by Trott** [comment, marked as answer]
By importing the plugin and setting that to `false`, I got this to work in both CLI and API.

In other words, instead of doing this:

```js
  ["remark-lint-no-auto-link-without-protocol", false],
```

I did this:

```js
import remarkLintNoAutoLinkWithoutProtocol from "remark-lint-no-auto-link-without-protocol";
```

...and then in the plugins array:

```js
  [remarkLintNoAutoLinkWithoutProtocol, false],
```

I suspect it would also work if I altered the preset array to remove the plugin before passing it to `use()`.

**Reply by Trott** [reply]
Confirmed that this also works:

```js
remarkPresetLintRecommended.plugins = remarkPresetLintRecommended.plugins.filter((fn) => fn.name !== 'remark-lint:no-auto-link-without-protocol');
```

Seems a bit wrong to modify it like that, but if the plugin is likely to be removed from the recommended preset at some point, I'm certainly OK with doing that until that happens.

**Reply by wooorm** [reply]
ah, yeah. `unified` itself doesn’t include any code to load things, because it can also be used in browsers. The string form is an engine (so the CLI) “overload”, where it’s mainly useful because of configuration from package.json or in YAML which can’t import.

I recommend using an `import` if you’re doing it in the node preset. But filtering might also work for your case.

> but if the plugin is likely to be removed from the recommended preset at some point, I'm certainly OK with doing that until that happens.

Yeah, I think I can do that in a minor because the plugin is most likely wrong and most unlikely useful 🤔
